### PR TITLE
[@KotlinCleanup] file AutoSizeCheckBoxPreference.kt

### PR DESCRIPTION
**Pull request by Jatin887**
[KotlinCleanup] file AutoSizeCheckBoxPreference.kt

## Purpose / Description
Kotlin cleanup of file AutoSizeCheckBoxPreference.kt

## Fixes
Part of #10489

## Approach
Using lint removed all the unnecessary error

## How Has This Been Tested?
By mannualy bulding the project
## Learning (optional, can help others)
_Describe the research stage_

_Links to blog posts, patterns, libraries or addons used to solve this problem_

## Checklist
_Please, go through these checks before submitting the PR._

- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

**Comment by Akshay0701** [comment]
@Jatin887 @UvrajSB both of you made pr on the same issue, 

while I definitely appreciate your efforts but before working on any issues, it's good practice to get that issue assigned to yourself and then make a pr.

**Comment by Jatin887** [comment]
@david-allison please review the pr

File: AnkiDroid/src/main/java/com/ichi2/ui/AutoSizeCheckBoxPreference.kt [hunk 1]
```diff
@@ -27,27 +27,28 @@ import com.ichi2.utils.ViewGroupUtils.getAllChildren
 // java.lang.ClassCastException: com.ichi2.ui.AutoSizeCheckBoxPreference cannot be cast to android.preference.Preference
 @Suppress("deprecation") // TODO Tracked in https://github.com/ankidroid/Anki-Android/issues/5019
 @KotlinCleanup("auto-lint (except unused class)")
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/ui/AutoSizeCheckBoxPreference.kt:29]: This wasn't removed

**Review by david-allison — COMMENTED** [review]
Seems good. Are some issues remaining? You haven't removed the annotation

File: AnkiDroid/src/main/java/com/ichi2/ui/AutoSizeCheckBoxPreference.kt (hunk 1 above)
Comment (Jatin887) [reply AnkiDroid/src/main/java/com/ichi2/ui/AutoSizeCheckBoxPreference.kt:29]: What the @Suppress("deprecation")?

Comment (Jatin887) [reply AnkiDroid/src/main/java/com/ichi2/ui/AutoSizeCheckBoxPreference.kt:29]: The annonation is required because otherwise it shows 
image

Comment (david-allison) [reply AnkiDroid/src/main/java/com/ichi2/ui/AutoSizeCheckBoxPreference.kt:29]: No, the KotlinCleanup

Comment (Jatin887) [reply AnkiDroid/src/main/java/com/ichi2/ui/AutoSizeCheckBoxPreference.kt]: removed now please check

File: AnkiDroid/src/main/java/com/ichi2/ui/AutoSizeCheckBoxPreference.kt [hunk 2]
```diff
@@ -13,41 +13,44 @@
...
-@Suppress("deprecation") // TODO Tracked in https://github.com/ankidroid/Anki-Android/issues/5019
-@KotlinCleanup("auto-lint (except unused class)")
-class AutoSizeCheckBoxPreference : android.preference.CheckBoxPreference {
-    @Suppress("unused") constructor(context: Context?) : super(context) {}
-    @Suppress("unused") constructor(context: Context?, attrs: AttributeSet?) : super(context, attrs) {}
-    @Suppress("unused") constructor(context: Context?, attrs: AttributeSet?, defStyleAttr: Int) : super(context, attrs, defStyleAttr) {}
-    @Suppress("unused") constructor(context: Context?, attrs: AttributeSet?, defStyleAttr: Int, defStyleRes: Int) : super(context, attrs, defStyleAttr, defStyleRes) {}
+// TODO Tracked in https://github.com/ankidroid/Anki-Android/issues/5019
+
+@Suppress("DEPRECATION")
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/ui/AutoSizeCheckBoxPreference.kt:32]: Why was this changed?

```suggestion
@Suppress("deprecation") // TODO Tracked in https://github.com/ankidroid/Anki-Android/issues/5019
```

File: AnkiDroid/src/main/java/com/ichi2/ui/AutoSizeCheckBoxPreference.kt [hunk 3]
```diff
@@ -13,41 +13,44 @@
...
+class AutoSizeCheckBoxPreference : CheckBoxPreference {
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/ui/AutoSizeCheckBoxPreference.kt:33]:
```suggestion
class AutoSizeCheckBoxPreference : android.preference.CheckBoxPreference {
```

**Review by david-allison — CHANGES_REQUESTED** [review]
Thanks! Probably the last round of fixes

File: AnkiDroid/src/main/java/com/ichi2/ui/AutoSizeCheckBoxPreference.kt (hunk 3 above)
Comment (Jatin887) [reply AnkiDroid/src/main/java/com/ichi2/ui/AutoSizeCheckBoxPreference.kt]: @david-allison all the changes are done . please review the pr

**Comment by david-allison** [comment]
One change in my requests remaining

File: AnkiDroid/src/main/java/com/ichi2/ui/AutoSizeCheckBoxPreference.kt [hunk 4]
```diff
@@ -13,41 +13,41 @@
...
-import com.ichi2.utils.KotlinCleanup
 import com.ichi2.utils.ViewGroupUtils.getAllChildren
 
 // extending androidx.preference didn't work:
 // java.lang.ClassCastException: com.ichi2.ui.AutoSizeCheckBoxPreference cannot be cast to android.preference.Preference
-@Suppress("deprecation") // TODO Tracked in https://github.com/ankidroid/Anki-Android/issues/5019
-@KotlinCleanup("auto-lint (except unused class)")
+// TODO Tracked in https://github.com/ankidroid/Anki-Android/issues/5019
+
+@Suppress("deprecation")
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/ui/AutoSizeCheckBoxPreference.kt:29]:
```suggestion
@Suppress("deprecation") // TODO Tracked in https://github.com/ankidroid/Anki-Android/issues/5019
```

Comment (Jatin887) [reply AnkiDroid/src/main/java/com/ichi2/ui/AutoSizeCheckBoxPreference.kt:29]: @david-allison is all changes done?
can i squash all commits?

Comment (david-allison) [reply AnkiDroid/src/main/java/com/ichi2/ui/AutoSizeCheckBoxPreference.kt:29]: we can, but it'd be ideal if you did first so the commit message is 100% relevant.

Comment (Jatin887) [reply AnkiDroid/src/main/java/com/ichi2/ui/AutoSizeCheckBoxPreference.kt:29]: ok

File: AnkiDroid/src/main/java/com/ichi2/ui/AutoSizeCheckBoxPreference.kt [hunk 5]
```diff
@@ -13,41 +13,41 @@
  You should have received a copy of the GNU General Public License along with
  this program.  If not, see <http://www.gnu.org/licenses/>.
  */
-package com.ichi2.ui
 
+package com.ichi2.ui
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/ui/AutoSizeCheckBoxPreference.kt:17]: Could you revert this change - not necessary

File: AnkiDroid/src/main/java/com/ichi2/ui/AutoSizeCheckBoxPreference.kt (hunk 4 above)
Comment (david-allison) [reply AnkiDroid/src/main/java/com/ichi2/ui/AutoSizeCheckBoxPreference.kt:29]: Just this one left

**Comment by Jatin887** [comment]
@david-allison squash all the commits please check

File: AnkiDroid/src/main/java/com/ichi2/ui/AutoSizeCheckBoxPreference.kt (hunk 4 above)
Comment (david-allison) [reply AnkiDroid/src/main/java/com/ichi2/ui/AutoSizeCheckBoxPreference.kt:29]: This is still pending

Comment (Jatin887) [reply AnkiDroid/src/main/java/com/ichi2/ui/AutoSizeCheckBoxPreference.kt]: is it solved in the new commit? so i can squash if it's done

Comment (Jatin887) [reply AnkiDroid/src/main/java/com/ichi2/ui/AutoSizeCheckBoxPreference.kt]: @david-allison i think all changes are done
Can i go ahead and squash all commits?

**Review by david-allison — APPROVED** [review]
Looks great, thanks!

**Review by mikehardy — APPROVED** [review]
Great thanks!

**Pull request merged**